provider/src/index.ts

```diff
@@ -36,56 +36,94 @@ export default {
           'access-control-allow-methods': 'POST',
           'access-control-allow-headers': 'content-type',
           'access-control-max-age': '1800',
-          'access-control-allow-origin': request.headers.get('Origin'),
+          'access-control-allow-origin': '*',
           vary: 'Origin',
           'access-control-allow-credentials': 'true',
         },
       })
     }
 
+    const url = new URL(request.url)
+    const { pathname } = url
+    const proxyURL = url.searchParams.get('url')
+
+    if (pathname === '/proxy' && proxyURL) {
+      const endpoint = new URL(proxyURL)
+      // Proxy the request
+      const response = await fetch(endpoint.toString(), {
```

Comment:
I am a bit worried this might be abused because right now it allows for _any_ kind of URL.